app/helpers/puppetca_helper.rb

```diff
@@ -1,8 +1,19 @@
 module PuppetcaHelper
+  STATES = [N_('revoked'), N_('pending'), N_('valid')]
+
   def state_filter
-    states = [N_('revoked'), N_('pending'), N_('valid')]
-    select_tag "Filter", options_for_select([''] + states.map { |s| [_(s), s] }, params[:state]), :class => "puppetca-filters",
-               :onchange => "window.location.href = '#{smart_proxy_puppetca_index_path(@proxy)}' + (this.value == '' ? '' : ('?state=' + this.value))"
+    select_tag "Filter", options_for_select([[_('valid or pending'), _('valid')+'|'+_('pending')]] +
+                                            STATES.map{|s| _(s)} +
+                                            [[_('all'),'']]),
+               :class => "puppetca-filters",
+               :onchange => "filterCerts(this.value)"
+  end
+
+  def state_counters
+    STATES.map do |state|
+      content_tag(:div, _(state)+":", :class => 'col-md-2')+
+        content_tag(:div, @certificates.count{|c| c.state == state}.to_s, :class => 'col-md-10')
```

Comment:
Even better, made this into patternfly cards and changed the order so pending is the first displayed, I think it should be obvious enough.